index.js

```diff
@@ -1,59 +1,108 @@
 'use strict';
 
 const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor'; // Key in annotations object that maps to an executor NPM module
+const SCHEMA_START = require('screwdriver-data-schema').plugins.queue.start;
 const Executor = require('screwdriver-executor-base');
+const Joi = require('joi');
+const Resque = require('node-resque');
+
+/**
+ * Strip the leading 'screwdriver-executor' from plugin names
+ * @method getPluginName
+ * @param {String} plugin
+ * @return {String}
+ */
+function getPluginName(plugin) {
+    return plugin.replace('screwdriver-executor-', '');
+}
 
-class ExecutorRouter extends Executor {
+class ExecutorQueue extends Executor {
     /**
      * Constructs a router for different Executor strategies.
      * @method constructor
      * @param  {Object}         config                      Object with executor and ecosystem
      * @param  {String}         [config.defaultPlugin]      Optional default executor
```

Comment:
This is where not caring what picks things up at the other end is nice. The worker(s) could ignore the attributes for the MVP. The works already been done here to send *something* for worker selection. I just feel that currently this PR does too much to that end.